pytorch_lightning/trainer/predict_loop.py

```diff
@@ -82,18 +114,44 @@ def predict_step(self, batch, batch_idx, dataloader_idx):
 
         self.trainer.call_hook("on_predict_batch_end", predictions, batch, batch_idx, dataloader_idx)
 
-        self._predictions[dataloader_idx].append(predictions)
+        if self.should_store_predictions:
+            self._predictions[dataloader_idx].append(predictions)
+
+    def _store_batch_indices(self, dataloader_idx: int) -> None:
+        batch_sampler = self.trainer.predict_dataloaders[dataloader_idx].batch_sampler
+        if isinstance(batch_sampler, IndexBatchSampler):
+            self.batch_indices = batch_sampler.batch_indices
+            if self.should_store_predictions:
+                self._batches_indices[dataloader_idx].append(batch_sampler.batch_indices)
+
+    def on_predict_start(self):
+        # enable eval mode + no grads
+        self.on_predict_model_eval()
+        self.trainer.lightning_module.zero_grad()
+        torch.set_grad_enabled(False)
+        # hook
+        self.trainer.call_hook("on_predict_start")
+
+    def on_predict_epoch_start(self):
+        # hook
+        self.trainer.call_hook("on_predict_epoch_start")
```

Comment:
I think I prefer calling

```python
self.call_hook("on_predict_epoch_start")
```

directly in the trainer if the function is going to be empty

Reply:
Fine :)